tests/p2p/test_server.py

```diff
@@ -29,25 +31,58 @@ def get_open_port():
 
 port = get_open_port()
 SERVER_ADDRESS = Address('127.0.0.1', udp_port=port, tcp_port=port)
-INITIATOR_PUBKEY = keys.PrivateKey(eip8_values['initiator_private_key']).public_key
+RECEIVER_PRIVKEY = keys.PrivateKey(eip8_values['receiver_private_key'])
+RECEIVER_PUBKEY = RECEIVER_PRIVKEY.public_key
+
+INITIATOR_PRIVKEY = keys.PrivateKey(eip8_values['initiator_private_key'])
+INITIATOR_PUBKEY = INITIATOR_PRIVKEY.public_key
 INITIATOR_ADDRESS = Address('127.0.0.1', get_open_port() + 1)
+
 INITIATOR_REMOTE = Node(INITIATOR_PUBKEY, INITIATOR_ADDRESS)
 
 
+def get_server(privkey, address, bootstrap_nodes=None, peer_class=DumbPeer):
+    if bootstrap_nodes is None:
+        bootstrap_nodes = []
+    chaindb = ChainDB(MemoryDB())
+    server = Server(
+        privkey,
+        address,
+        chaindb,
+        bootstrap_nodes,
+        1,
+        min_peers=1,
+        peer_class=peer_class,
+    )
+    return server
+
+
+async def send_auth_msg_to_server(address, messages):
+    reader, writer = await asyncio.open_connection(address.ip, address.tcp_port)
+    writer.write(messages)
+    await writer.drain()
+
+
 @pytest.fixture
 def server():
-    privkey = keys.PrivateKey(eip8_values['receiver_private_key'])
-    chaindb = ChainDB(MemoryDB())
-    server = Server(privkey, SERVER_ADDRESS, chaindb, [], 1)
+    server = get_server(RECEIVER_PRIVKEY, SERVER_ADDRESS, bootstrap_nodes=[], peer_class=ETHPeer)
     return server
 
 
+async def mock_do_p2p_handshake(self):
+    pass
+
+
 # FIXME: Instead of starting an actual server and send data over the wire, this test should create
 # StreamReaders and StreamWriters and connect them directly, like in test_auth.py. That way we
 # don't need those "strategically" placed sleep() calls.
 @pytest.mark.asyncio
 async def test_server_authenticates_incoming_connections(server, event_loop):
     await server.start()
+
+    # Only test the authentication in this test.
+    ETHPeer.do_p2p_handshake = mock_do_p2p_handshake
```

Comment:
This is not a good idea as any test that runs after this will use the mock method implementation as well. You should use https://docs.pytest.org/en/latest/monkeypatch.html instead